lib/connectors/mongodb/connector.rb

```diff
@@ -56,30 +70,166 @@ def initialize(configuration: {})
         @direct_connection = configuration.dig(:direct_connection, :value)
       end
 
-      def yield_documents
+      def yield_documents(job_description = {}, &on_doc_serialization)
```

Comment:
It's used in `call_db_function_on_collection`, when a document is serialized. We can also pass it implicitly, but I thought it would be useful to have it explicitly in the code.